cobalt/browser/metrics/cobalt_metrics_service_client.cc

```diff
@@ -0,0 +1,196 @@
+namespace metrics {
+
+// How frequently to attempt uploading.
+const int kStandardUploadIntervalSeconds = 5 * 60;  // 5 minutes.
+
+void CobaltMetricsServiceClient::SetOnUploadHandler(
+    h5vcc::MetricEventHandlerWrapper* metric_event_handler) {
+  metric_event_handler_ = metric_event_handler;
+  if (log_uploader_) {
+    log_uploader_->SetOnUploadHandler(metric_event_handler_);
+  }
+}
+
+CobaltMetricsServiceClient::CobaltMetricsServiceClient(
+    ::metrics::MetricsStateManager* state_manager, PrefService* local_state)
+    : metrics_state_manager_(state_manager) {
+  metrics_service_ = std::make_unique<::metrics::MetricsService>(
+      metrics_state_manager_, this, local_state);
+}
+
+::metrics::MetricsService* CobaltMetricsServiceClient::GetMetricsService() {
+  return metrics_service_.get();
+}
+
+ukm::UkmService* CobaltMetricsServiceClient::GetUkmService() {
+  // TODO(b/284467142): UKM disabled in Cobalt currently, replace when UKM
+  // is supported.
+  return nullptr;
+}
+
+void CobaltMetricsServiceClient::SetMetricsClientId(
+    const std::string& client_id) {
+  // TODO(b/286066035): What to do with client id here?
+}
+
+int32_t CobaltMetricsServiceClient::GetProduct() {
+  // Note, Product is a Chrome concept and similar dimensions will get logged
+  // elsewhere downstream. This value doesn't matter.
+  return 0;
+}
+
+std::string CobaltMetricsServiceClient::GetApplicationLocale() {
+  // The locale will be populated by the web client, so return value is
+  // inconsequential.
+  return "en-US";
+}
+
+bool CobaltMetricsServiceClient::GetBrand(std::string* brand_code) {
+  // "false" means no brand code available. We set the brand when uploading
+  // via GEL.
+  return false;
+}
+
+::metrics::SystemProfileProto::Channel
+CobaltMetricsServiceClient::GetChannel() {
+  // We aren't Chrome and don't follow the same release channel concept.
+  // Return value here is unused in downstream logging.
+  return ::metrics::SystemProfileProto::CHANNEL_UNKNOWN;
+}
+
+std::string CobaltMetricsServiceClient::GetVersionString() {
+  // All version strings will be attached to the GEL event sent by the web
+  // client, so the value doesn't matter much here.
+  return "1.0";
+}
+
+void CobaltMetricsServiceClient::OnEnvironmentUpdate(
+    std::string* serialized_environment) {
+  // Environment updates are serialized SystemProfileProto changes. All
+  // system info should be reported with the web client, so this is a no-op.
+}
+
+void CobaltMetricsServiceClient::CollectFinalMetricsForLog(
+    const base::Closure& done_callback) {
+  // Any hooks that should be called before each new log is uploaded, goes here.
+  // Chrome uses this to update memory histograms. Regardless, you must call
+  // done_callback when done else the uploader will never get invoked.
+  std::move(done_callback).Run();
+}
+
+std::string CobaltMetricsServiceClient::GetMetricsServerUrl() {
+  // Cobalt doesn't upload anything itself, so any URLs are no-ops.
+  return "";
```

Comment:
Should this perhaps be NOTIMPLEMENTED to ensure it's not getting called ?

Reply:
It actually does get called, we just don't use it. See reporting_service: http://cs/cobalt/components/metrics/reporting_service.h;l=86;rcl=909d97218335a5fc69d7861448d8ac53deb44a97